### PR TITLE
feat: Implement a tick box for organizers/admin to override Email

### LINKED ISSUE #2953: Implement a tick box for organizers/admin to override Email requirement
When a speaker registers they have to enter an email. It must be possible to override this requirement for organizers/admins, e.g. if a VIP speaker does not want to sign up himself. Please implement a small tick box below the Email Field as follows. Only show this option for organizers/admins (not for speaker who submit this through the online form):
* [ ] Do not require email for this speaker.

Screenshot from 2019-05-18 19-29-21

### PR DESCRIPTION
**Pull request by shreyanshdwivedi**
Fixes #2953

#### Short description of what this resolves:
When a speaker registers they have to enter an email. It must be possible to override this requirement for organizers/admins

Screenshot_2019-08-03 Create session Speakers cfs Events Open Event

Screenshot_2019-08-03 Create session Sessions cfs Events Open Event

#### Changes proposed in this pull request:
- Adds a checkbox to override email while creating speaker from event dashboard

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/controllers/events/view/sessions/create.js [hunk 1]
```diff
@@ -6,6 +6,9 @@ export default Controller.extend({
       await this.get('model.session').save();
       if (this.addNewSpeaker) {
         let newSpeaker = this.get('model.speaker');
+        if (newSpeaker.isEmailOverridden) {
+          newSpeaker.set('email', this.authManager.currentUser.email);
+        }
```
Comment (mrsaicharan1) [inline app/controllers/events/view/sessions/create.js]: As this code is repetitive, is it possible to shift it to a mixin?

Comment (shreyanshdwivedi) [reply app/controllers/events/view/sessions/create.js]: I think that these 2 times are should not be considered for any mixin or helper. I think it's fine

**Comment by shreyanshdwivedi** [comment]
@CosmicCoder96 @uds5501 please review

**Comment by shreyanshdwivedi** [comment]
@kushthedude @CosmicCoder96 pleaae review

**Pull request merged**